warrior/Framework/ClassUtils/WSelenium/browser_mgmt.py

```diff
@@ -539,6 +545,15 @@ def _make_ff(self, webdriver_remote_url, desired_capabilites, profile_dir, **kwa
                 # And firefox will need to know how to handle it
                 ff_capabilities['acceptInsecureCerts'] = True
 
+                if binary not in [False, None]:
+                    if not fileExists(binary):
+                        print_warning("Given firefox binary '{}' does not exist, default "
+                                      "firefox will be used for execution.".format(binary))
+                        binary = None
+                else:
+                    print_info("No value given for firefox binary, default "
+                               "firefox will be used for execution.")
+
                 # Force disable marionette, only needs in Selenium 3 with FF ver < 47
                 # Without these lines, selenium may encounter capability not found issue
                 # https://github.com/seleniumhq/selenium/issues/2739
```

Comment:
Hi @sanikakulkarni 

Query:
-------
If we do not support Windows, it would affect the testing of Nagammai's team testing activities as they have been using both Linux and Windows for their test automation.
As per my knowledge, we do not have an OS independent command to get the firefox version.

Suggestion:
-------------
we could provide support, just for the version command update.
the following code would address this issue:

def get_firefox_version(self, binary):
        """
            Use firefox binary to find out firefox version
            before launching firefox in selenium
        """
        import platform
        raw_version=""
                   
        if platform.system() in "Linux":
            if binary in [False, None]:
                binary = "firefox"
            raw_version = check_output([binary, "-v"]) 
        elif platform.system() in "Windows":
            if binary in [False, None]:
                binary= FirefoxBinary._default_windows_location()
            print_info("*** ff binary for windows is : {} ".format(binary))
            command = "%s -v | more" % (binary)    
            raw_version = check_output(command)          
                
        version = False
        try:
            #raw_version = check_output(ff_version)
            print_info("*** Raw_version is: {} ".format(raw_version))
            match = re.search(r"\d+\.\d+", raw_version)
            if match is not None:
                version = LooseVersion(match.group(0))
            else:
                print_info("Cannot parse Firefox version: {}".format(raw_version))
        except CalledProcessError:
            print_error("Cannot find firefox version, will not launch browser")
        return version

Reply:
Hi @safarheen,

This code looks fine. Please add it to the codebase (py2 and py3. Please note that the code *may* require changes when being ported to py3) and rerun the selenium projects and attach them to the JIRA ticket. Thanks.

Reply:
Hi @sanikakulkarni 

I have added the platform check in the code for both py and py3. attached the tests in JIRA.
Please approve and merge the PRs